misc/scripts/install_linuxbsd_deps.sh

```diff
@@ -0,0 +1,143 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
+
+print_info() {
+	echo -e "\e[1m[*] $1\e[0m"
+}
+
+print_success() {
+	echo -e "\e[1;92m[*] $1\e[0m"
+}
+
+print_warning() {
+	echo -e "\e[1;93m[!] Warning: $1\e[0m"
+}
+
+print_error() {
+	echo -e "\e[1;91m[!] Error: $1\e[0m"
+}
+
+# The detection below also covers derivatives such as Linux Mint and Manjaro.
+if [[ -f "/etc/alpine-release" ]]; then
+	distribution="alpine"
+elif [[ -f "/etc/arch-release" ]]; then
+	distribution="arch"
+elif [[ -f "/etc/lsb-release" ]]; then
+	# Ubuntu must be checked before checking for Debian as Ubuntu also has `/etc/debian_version`.
+	distribution="ubuntu"
+	ubuntu_version_year="$(grep DISTRIB_RELEASE /etc/lsb-release | cut -d = -f 2 | cut -d . -f 1)"
+	ubuntu_version_month="$(grep DISTRIB_RELEASE /etc/lsb-release | cut -d = -f 2 | cut -d . -f 2)"
+	# Ubuntu 18.04 is required for out-of-the-box support.
+	if [[ "$ubuntu_version_year" -lt 18 ]]; then
+		print_warning "Your Ubuntu version ($ubuntu_version_year.$ubuntu_version_month) does not include GCC 7 or later, which is required to compile Godot.\n    You'll have to install a more recent GCC version using a third-party PPA, or upgrade to a newer Ubuntu version.\n    PPA: https://launchpad.net/~ubuntu-toolchain-r/+archive/ubuntu/test"
+	fi
+elif [[ -f "/etc/debian_version" ]]; then
+	distribution="debian"
+	debian_version="$(cut -d "." -f 1 </etc/debian_version)"
+	# Debian 11 (bullseye) is required for out-of-the-box support.
+	if [[ "$debian_version" -lt 11 ]]; then
+		print_warning "Your Debian version ($debian_version) does not include GCC 7 or later, which is required to compile Godot.\n    You'll have to install a more recent GCC version using a third-party repository, or upgrade to a newer Debian version."
+	fi
+elif [[ -f "/etc/fedora-release" ]]; then
+	distribution="fedora"
+elif [[ -f "/etc/gentoo-release" ]]; then
+	distribution="gentoo"
+elif [[ -f "/etc/mageia-release" ]]; then
+	# TODO: Is this the correct way to detect Mageia?
```

Comment:
That works yeah:
```
$ cat /etc/mageia-release 
Mageia release 8 (Cauldron) for x86_64
```

But the script itself won't work as Mageia also has `/etc/lsb-release`, and would thus be found as "Ubuntu". I think many other distros have `lsb-release`, which is actually the standard way for LSB-compliant distros to document their versioning:
```
$ cat /etc/lsb-release
LSB_VERSION=
DISTRIB_ID="Mageia"
DISTRIB_RELEASE=8
DISTRIB_CODENAME=mga8
DISTRIB_DESCRIPTION="Mageia 8"
```

Reply:
For the reference on a Mageia 8 system:
```
$ ls -l /etc/*-release
-rw-r--r-- 1 root root 104 Feb 15  2020 /etc/lsb-release
-rw-r--r-- 1 root root  39 Nov  3 12:32 /etc/mageia-release
lrwxrwxrwx 1 root root  14 Nov  3 12:32 /etc/mandrakelinux-release -> mageia-release
lrwxrwxrwx 1 root root  14 Nov  3 12:32 /etc/mandrake-release -> mageia-release
lrwxrwxrwx 1 root root  14 Nov  3 12:32 /etc/mandriva-release -> mageia-release
lrwxrwxrwx 1 root root  21 Nov  3 12:32 /etc/os-release -> ../usr/lib/os-release
lrwxrwxrwx 1 root root  14 Nov  3 12:32 /etc/redhat-release -> mageia-release
lrwxrwxrwx 1 root root  14 Nov  3 12:32 /etc/system-release -> mageia-release
```